### PR TITLE
Add flow to Scopes.js, WhyPaused.js, SourcesTree.js,...

### PR DESCRIPTION
**Pull request by arthur801031**
Associated Issue: #1894 

### Summary of Changes
Add flow to the following:
- SecondaryPanes/Scopes.js
- SecondaryPanes/WhyPaused.js
- SecondaryPanes/index.js
- SecondaryPanes/SourcesTree.js
- SecondaryPanes/WelcomeBox.j
- shared/Accordion.js

### Test Plan
- Passed `yarn run flow`

### CODE REVIEW

File: src/components/SecondaryPanes/Expressions.js [hunk 1]
```diff
@@ -52,6 +50,8 @@ const Expressions = React.createClass({
     loadedObjects: ImPropTypes.map.isRequired
   },
 
+  _input: (null: any),
```
Comment (arthur801031) [inline src/components/SecondaryPanes/Expressions.js]: I moved it here to fix
screen shot 2017-02-18 at 10 21 41 am

File: src/components/SourcesTree.js [hunk 2]
```diff
@@ -17,6 +18,15 @@ const { showMenu } = require("./shared/menu");
 const { copyToTheClipboard } = require("../utils/clipboard");
 const { throttle } = require("../utils/utils");
 
+type CreateTree = {
```
Comment (arthur801031) [inline src/components/SourcesTree.js]: Should we create types for `focusedItem`, `parentMap`, `sourceTree`,...? Or is using `any` ok for now?

File: src/components/shared/Accordion.js [hunk 3]
```diff
@@ -6,19 +7,32 @@ const Svg = require("./Svg");
...
 const Accordion = React.createClass({
   propTypes: {
-    items: PropTypes.array
+    items: PropTypes.array.isRequired
   },
 
   displayName: "Accordion",
 
-  getInitialState: function() {
+  getInitialState() {
```
Comment (arthur801031) [inline src/components/shared/Accordion.js]: I removed `function` to keep the style consistent.

File: src/components/shared/Accordion.js [hunk 4]
```diff
@@ -6,19 +7,32 @@ const Svg = require("./Svg");
...
     return { opened: this.props.items.map(item => item.opened),
       created: [] };
   },
 
-  componentWillReceiveProps: function(nextProps) {
+  componentWillReceiveProps(nextProps: Props) {
```
Comment (arthur801031) [inline src/components/shared/Accordion.js]: This is strange. flow does not complain about missing annotation for `nextProps`in `src/components/SourcesTree.js`'s `componentWillReceiveProps`, but it does for this file.
screen shot 2017-02-20 at 2 49 39 pm

File: src/components/SecondaryPanes/index.js [hunk 5]
```diff
@@ -30,6 +31,24 @@ const Accordion = React.createFactory(require("../shared/Accordion"));
 const CommandBar = React.createFactory(require("./CommandBar").default);
 require("./SecondaryPanes.css");
 
+type StartItems = {
```
Comment (clarkbw) [inline src/components/SecondaryPanes/index.js]: Can you explain the difference here between `StartItems` and `EndItems`? they look so similar except for `button` vs `buttons` and I'm not sure why we have a `button` attribute.

File: src/components/SecondaryPanes/index.js [hunk 6]
```diff
@@ -121,10 +140,11 @@ const SecondaryPanes = React.createClass({
...
   getStartItems() {
-    const scopesContent = this.props.horizontal ? this.getScopeItem() : null;
+    const scopesContent: any = this.props.horizontal ?
+      this.getScopeItem() : null;
     const isPaused = () => !!this.props.pauseData;
 
-    const items = [
+    const items: Array<StartItems> = [
       { header: L10N.getStr("breakpoints.header"),
         buttons: this.renderBreakpointsToggle(),
```
Comment (clarkbw) [inline src/components/SecondaryPanes/index.js]: Maybe this is what confuses me.  These are `StartItems` but there is a `buttons` attribute in the first one.

File: src/components/SourcesTree.js (hunk 2 above)
Comment (clarkbw) [inline src/components/SourcesTree.js]: I think `any` is ok for now.  This is a good size change and we can dig into that later.

File: src/components/SecondaryPanes/index.js (hunk 5 above)
Comment (arthur801031) [inline src/components/SecondaryPanes/index.js]: Hi @clarkbw 

I totally got that wrong. Sorry 😅. I'll push a fix right now. Thank you!

File: src/components/shared/Accordion.js (hunk 4 above)
Comment (clarkbw) [inline src/components/shared/Accordion.js]: I'm not sure why this is.

Comment (arthur801031) [inline src/components/shared/Accordion.js]: I did some investigation. It seems files in the `shared` folder that have `componentWillReceiveProps `will have this error if we don't have a type for `nextProps`.

Those files are `Accordion.js` and `ManagedTree.js`.

Comment (jasonLaster) [inline src/components/shared/Accordion.js]: strange

**Pull request merged**